src/coreclr/System.Private.CoreLib/src/Microsoft/Win32/OAVariantLib.cs

```diff
@@ -32,96 +36,175 @@ internal static unsafe partial class OAVariantLib
         public const int CalendarHijri = 0x08;
         public const int LocalBool = 0x10;
 
-        internal static readonly Type?[] ClassTypes = {
-            typeof(Empty),
-            typeof(void),
-            typeof(bool),
-            typeof(char),
-            typeof(sbyte),
-            typeof(byte),
-            typeof(short),
-            typeof(ushort),
-            typeof(int),
-            typeof(uint),
-            typeof(long),
-            typeof(ulong),
-            typeof(float),
-            typeof(double),
-            typeof(string),
-            typeof(void),
-            typeof(DateTime),
-            typeof(TimeSpan),
-            typeof(object),
-            typeof(decimal),
-            null,  // Enums - what do we do here?
-            typeof(Missing),
-            typeof(DBNull),
+        private static readonly Dictionary<Type, VarEnum> ClassTypes = new Dictionary<Type, VarEnum>
```

Comment:
I don't see currency or null in this list. How are we handling those?

Reply:
Currency: The previous ClassTypes doesn't contain an entry for `CurrencyWrapper` or something. It's not mapped.
Null: The misplace mentioned in https://github.com/dotnet/runtime/pull/100176#issuecomment-2018167832 cause it to be mapped to CV_MISSING, and rejected in `OAVaraint::CVToVT`.